src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -2964,10 +2964,11 @@ private RefKind GetEffectiveParameterRefKind(ParameterSymbol parameter, RefKind
                 }
                 else
                 {
+                    bool isThisArgument = arguments.IsExtensionMethodThisArgument(argumentPosition);
                     RefKind parameterRefKind = parameters.ParameterRefKinds.IsDefault ? RefKind.None : parameters.ParameterRefKinds[argumentPosition];
-                    conversion = CheckArgumentForApplicability(candidate, argument, argumentRefKind, parameters.ParameterTypes[argumentPosition], parameterRefKind, ignoreOpenTypes, ref useSiteDiagnostics);
+                    conversion = CheckArgumentForApplicability(candidate, argument, argumentRefKind, parameters.ParameterTypes[argumentPosition], parameterRefKind, ignoreOpenTypes, isThisArgument, ref useSiteDiagnostics);
 
-                    if (arguments.IsExtensionMethodThisArgument(argumentPosition) && !Conversions.IsValidExtensionMethodThisArgConversion(conversion))
+                    if (isThisArgument && !Conversions.IsValidExtensionMethodThisArgConversion(conversion))
```

Comment:
Added tests for identity, implicit, and operator conversions for both ref and ref-readonly extension methods.

---
In reply to: [120224648](https://github.com/dotnet/roslyn/pull/19703#discussion_r120224648) [](ancestors = 120224648)